utils/bump_version.py

```diff
@@ -120,7 +124,7 @@ def finalize_release_date(self) -> None:
             f.write('=' * len(heading) + '\n')
             f.write(self.filter_empty_sections(body))
 
-    def add_release(self, version_info) -> None:
+    def add_release(self, version_info) -> None:  # type: ignore[no-untyped-def]
```

Comment:
I think we can have:

```python
VersionInfoType = Union[
    tuple[int, int, int, str], 
    tuple[int, int, int, str, int]
]
```

Reply:
i've added this alias, however there's a couple of places where the code assumes that version_info[4] exists. I've added ignore statements for these for now, but it will need a more careful look

Reply:
> where the code assumes that version_info[4] exists.

Oh, then you can maybe remove the union (actually, I incorrectly counted my elements..., I don't think we should have the first `tuple[int, int, int, str]`)